src/sentry/models/orgauthtoken.py

```diff
@@ -0,0 +1,64 @@
+from __future__ import annotations
+
+from django.core.exceptions import ValidationError
+from django.db import models
+from django.utils import timezone
+from django.utils.encoding import force_text
+
+from sentry.conf.server import SENTRY_SCOPES
+from sentry.db.models import (
+    ArrayField,
+    BaseManager,
+    FlexibleForeignKey,
+    Model,
+    control_silo_only_model,
+    sane_repr,
+)
+from sentry.db.models.fields.hybrid_cloud_foreign_key import HybridCloudForeignKey
+
+
+def validate_scope_list(value):
+    for choice in value:
+        if choice not in SENTRY_SCOPES:
+            raise ValidationError(f"{choice} is not a valid scope.")
+
+
+@control_silo_only_model
+class OrgAuthToken(Model):
+    __include_in_export__ = True
+
+    organization = HybridCloudForeignKey("sentry.Organization", null=False, on_delete="cascade")
+    # max JWT token length is 2048, +7 for our custom prefix (sntrys_)
+    token = models.CharField(max_length=2055, unique=True, null=False)
+    name = models.CharField(max_length=255, null=False)
+    scope_list = ArrayField(
+        models.TextField(),
+        validators=[validate_scope_list],
+    )
+
+    created_by = FlexibleForeignKey("sentry.User", null=True, blank=True)
+    date_added = models.DateTimeField(default=timezone.now, null=False)
+    date_last_used = models.DateTimeField(null=True, blank=True)
+    project_last_used = HybridCloudForeignKey(
+        "sentry.Project", null=True, blank=True, on_delete="cascade"
+    )
```

Comment:
Are you sure you want cascading here? If the last used project is deleted should the org token be deleted?

Reply:
Very good point @markstory, I missed this.

Reply:
@mydea we can just put the project id here as a `BoundedBigIntegerField` and forget about possible accidental cascading deletions.